### ISSUE TITLE
How to use zbxdb config file

### ISSUE THREAD

**Issue by starlessboi** [ikzelf/zbxdb#4]
Hi
I tried to use oracle sample config file, as below, but faced problem (I used my url ...).
[zbxdb]
db_url = //localhost/fsdb01
username = postgres
password =
db_type = oracle
db_driver = cx_Oracle
instance_type = rdbms
role = normal
out_dir = $HOME/zbxora_out
hostname = testhost
checks_dir = etc/zbxdb_checks
site_checks = sap,ebs
password_enc = cjB0cjAwdA==

a few questions
why user name is postgres...
what is password_enc? what should I fill in it?
when I entered my password in password field and user (instead of postgres), I got following error:
Traceback (most recent call last):
  File "/usr/local/bin/zbxdb.py", line 159, in <module>
    config = get_config(ARGS.configfile)
  File "/usr/local/bin/zbxdb.py", line 132, in get_config
    enc = encrypted(config['password'])
  File "/usr/local/bin/zbxdb.py", line 59, in encrypted
    return base64.b64encode(bytes(plain, 'utf-8'))
TypeError: str() takes at most 1 argument (2 given)

**Comment by ikzelf** [comment]
Nice  catch ….. it’s a missing doc I think.
1) You can pick the username you normally use for your monitoring, probably the same a in zbxora.
2) the password, you enter  in plain text in password
3) at startup zbxdb ‘encrypts’ the password to password_enc and uses this for the rest of time.
4) so you keep password_enc empty, or remove it completely and it will magically return and password will disappear.

**Comment by starlessboi** [comment]
[zbxdb]
db_url = xx.xx.xx.xx/dbs
username = dbmng
password = mypwsd
db_type = oracle
db_driver = cx_Oracle
instance_type = rdbms
role = normal
out_dir = /etc/zabbix/scripts/zbxdb_out
hostname = dbs_host
checks_dir = /etc/zabbix/scripts/zbxdb_checks
site_checks = sap,ebs
password_enc =

Hi again, Thanks, I used what you told as above config, but the error is the same, what is wrong with my config file

**Comment by ikzelf** [comment]
Using  your file (I called it z.cfg) I got:
$>bin/zbxdb.py -c z.cfg
2019-01-15 14:13:21 start python-3.6.4 zbxdb-0.77 pid=23302 Connecting for hostname dbs_host...
2019-01-15 14:13:21 first encrypted the plaintext password and removed from config
2019-01-15 14:13:21.743105 zbxdb found db_type=oracle, driver cx_Oracle; checking for driver
<module 'cx_Oracle' from '/Users/ronr/.pyenv/versions/3.6.4/lib/python3.6/site-packages/cx_Oracle.cpython-36m-darwin.so'>
2019-01-15 14:13:21.746353 zbxdb driver cx_Oracle loaded
2019-01-15 14:13:21.746747 zbxdb driver drivererrors for cx_Oracle loaded
2019-01-15 14:13:21.747134 zbxdb dbconnections for oracle loaded
<module 'dbconnections.oracle' from '/Users/ronr/Documents/git/zbxdb/bin/dbconnections/oracle.py'>
<module 'drivererrors.cx_Oracle' from '/Users/ronr/Documents/git/zbxdb/bin/drivererrors/cx_Oracle.py'>
2019-01-15 14:13:21.747190 site_checks: sap,ebs
2019-01-15 14:13:21.747201 out_file:/etc/zabbix/scripts/zbxdb_out/z.zbx
Traceback (most recent call last):
  File "bin/zbxdb.py", line 259, in <module>
    conn = dbc.connect(dbdr, config)
  File "/Users/ronr/Documents/git/zbxdb/bin/dbconnections/oracle.py", line 55, in connect
    x = db.connect(connect_string(c), mode=c['omode'])
cx_Oracle.DatabaseError: ORA-12154: TNS:could not resolve the connect identifier specified

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "bin/zbxdb.py", line 593, in <module>
    to_outfile(config, ME + "[connect,status]", ecode)
  File "bin/zbxdb.py", line 48, in to_outfile
    c['OUTF'] = open(c['out_file'], "w")
FileNotFoundError: [Errno 2] No such file or directory: '/etc/zabbix/scripts/zbxdb_out/z.zbx’

zbxdb first ‘encrypts’ the password to password_enc
Then it tries to connect to a nonexisting url, OK
problem is that you try to write in a directory that does not exist.

When the connection is going to succeed, the next error will be that it can’t find the site checks (sap and ebs are just example)

I made zbxdb such that it can run with a very  regular account. No root no dba,  just a regular user with regular privileges.
So normally I would keep the zbx_out directory in the home directory of the user that runs zbxdb.
It could be at any location but has to pre-exist.
